sdk/include/ddk/d3dkmddi.h

```diff
@@ -0,0 +1,239 @@
+            UINT DXGK_ALLOC_RESERVED11   : 1;
+            UINT DXGK_ALLOC_RESERVED10   : 1;
+            UINT DXGK_ALLOC_RESERVED9    : 1;
+            UINT DXGK_ALLOC_RESERVED4    : 1;
+            UINT DXGK_ALLOC_RESERVED3    : 1;
+            UINT DXGK_ALLOC_RESERVED2    : 1;
+            UINT DXGK_ALLOC_RESERVED1    : 1;
+            UINT DXGK_ALLOC_RESERVED0    : 1;
+        };
+        UINT Value;
+    };
+} DXGK_ALLOCATIONINFOFLAGS;
+C_ASSERT(sizeof(DXGK_ALLOCATIONINFOFLAGS) == 0x4);
+
+typedef struct _DXGK_ALLOCATIONUSAGEINFO1
+{
+    union
+    {
+        struct
+        {
+            UINT PrivateFormat  : 1;
+            UINT Swizzled       : 1;
+            UINT MipMap         : 1;
+            UINT Cube           : 1;
+            UINT Volume         : 1;
+            UINT Vertex         : 1;
+            UINT Index          : 1;
+            UINT Reserved       : 25;
+        };
+        UINT Value;
+    } Flags;
+    union
+    {
+        D3DDDIFORMAT Format;
+        UINT         PrivateFormat;
+    };
+    UINT SwizzledFormat;
+    UINT ByteOffset;
+    UINT Width;
+    UINT Height;
+    UINT Pitch;
+    UINT Depth;
+    UINT SlicePitch;
+} DXGK_ALLOCATIONUSAGEINFO1;
+C_ASSERT(sizeof(DXGK_ALLOCATIONUSAGEINFO1) == 0x24);
+
+typedef struct _DXGK_ALLOCATIONUSAGEHINT
+{
+    UINT                      Version;
+    DXGK_ALLOCATIONUSAGEINFO1 v1;
+} DXGK_ALLOCATIONUSAGEHINT;
+C_ASSERT(sizeof(DXGK_ALLOCATIONUSAGEHINT) == 0x28);
+
+typedef struct _DXGK_ALLOCATIONINFO
+{
+    VOID*                      pPrivateDriverData;
+    UINT                       PrivateDriverDataSize;
+    UINT                       Alignment;
+    SIZE_T                     Size;
+    SIZE_T                     PitchAlignedSize;
+    DXGK_SEGMENTBANKPREFERENCE HintedBank;
+    DXGK_SEGMENTPREFERENCE     PreferredSegment;
+    UINT                       SupportedReadSegmentSet;
+    UINT                       SupportedWriteSegmentSet;
+    UINT                       EvictionSegmentSet;
+    union
+    {
+        UINT MaximumRenamingListLength;
+        UINT PhysicalAdapterIndex;
+    };
+    HANDLE hAllocation;
+    union
+    {
+        DXGK_ALLOCATIONINFOFLAGS Flags;
+#if (DXGKDDI_INTERFACE_VERSION >= DXGKDDI_INTERFACE_VERSION_WDDM2_0)
+        DXGK_ALLOCATIONINFOFLAGS_WDDM2_0 FlagsWddm2;
+#endif
+    };
+    DXGK_ALLOCATIONUSAGEHINT* pAllocationUsageHint;
+    UINT                      AllocationPriority;
+#if (DXGKDDI_INTERFACE_VERSION >= DXGKDDI_INTERFACE_VERSION_WDDM3_0)
+    DXGK_ALLOCATIONINFOFLAGS2 Flags2;
+#endif
+} DXGK_ALLOCATIONINFO;
+#ifdef _WIN64
+#if (DXGKDDI_INTERFACE_VERSION >= DXGKDDI_INTERFACE_VERSION_WDDM3_0)
+C_ASSERT(sizeof(DXGK_ALLOCATIONINFO) == 0x5C);
+#else
+C_ASSERT(sizeof(DXGK_ALLOCATIONINFO) == 0x58);
+#endif
+#else
+#if (DXGKDDI_INTERFACE_VERSION >= DXGKDDI_INTERFACE_VERSION_WDDM3_0)
+C_ASSERT(sizeof(DXGK_ALLOCATIONINFO) == 0x40);
+#else
+C_ASSERT(sizeof(DXGK_ALLOCATIONINFO) == 0x3C);
+#endif
+#endif
+#endif // _D3DKMDDI_H_
```

Comment:
```suggestion

#endif // _D3DKMDDI_H_
```
(breathing separating space with the other endifs)